### PR TITLE
Added the writing-recipes topic

### PR DESCRIPTION
**Pull request by jimfulton**
Also: 

- De-document install args in help https://github.com/buildout/buildout/issues/358

- Improved help for init (just noticed it needed love when looking at install :))

### CODE REVIEW

File: doc/topics/writing-recipes.rst [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+
+- If an error is raised, it is the responsibility of the recipe to
+  undo any partial changes.
+
+- If the recipe created any files or directories, the recipe should
+  return their paths.  Doing so allows Buildout to take care of
+  removing them if the part is uninstalled, making a separate
+  uninstall recipe unnecessary.
+
+To make these responsibilities easier to cope with, the ``option``
```
Comment (mgedmin) [inline doc/topics/writing-recipes.rst]: Is this the same as the `options` object, which was mentioned as the 3rd argument passed to the constructor?

File: doc/topics/writing-recipes.rst [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+      <logfile>
+        path /sample/parts/server/transcript.log
+      </logfile>
+    </eventlog>
+
+.. -> server
+
+   >>> server = server.replace('/sample', os.getcwd())
+
+Here's the recipe source, ``src/democonfigrecipe.py``::
```
Comment (mgedmin) [inline doc/topics/writing-recipes.rst]: You probably want that fancy `.. code-block:: python` directive here, to get syntax highlighting.

File: doc/topics/writing-recipes.rst [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+Initialization phase: the constructor
+-------------------------------------
+
+The constructor is passed 3 arguments:
+
+``buildout``
+   The buildout configuration
+
+   The buildout configuration is a mapping from section names to
+   sections.  Sections, are mappings from option names to values.  The
```
Comment (jimfulton) [inline doc/topics/writing-recipes.rst]: Thanks, fixed.

File: doc/topics/writing-recipes.rst (hunk 2 above)
Comment (jimfulton) [inline doc/topics/writing-recipes.rst]: Python is the default language, so don't need it and don't want the clutter. :)

File: doc/topics/writing-recipes.rst (hunk 1 above)
Comment (jimfulton) [inline doc/topics/writing-recipes.rst]: Yup, as is noted on the next line.

**Pull request merged**